extensions/amp-story/0.1/amp-story-page.js

```diff
@@ -288,24 +288,20 @@ export class AmpStoryPage extends AMP.BaseElement {
    * @public
    */
   calculateLoadStatus() {
-    if (this.isLoaded_ || this.pageElements_.length == 0) {
+    const visiblePageElements = this.pageElements_.filter(pageElement =>
+      !pageElement.isHiddenByMediaQuery());
+
+    if (this.isLoaded_ || visiblePageElements.length == 0) {
       return true;
     }
 
-    let isPageLoaded = true;
-    let canPageBeShown = false;
-
-    this.pageElements_.forEach(pageElement => {
-      pageElement.updateState();
+    visiblePageElements.forEach(pageElement => pageElement.updateState());
 
-      if (isPageLoaded) {
-        isPageLoaded = pageElement.isLoaded || pageElement.hasFailed;
-      }
+    const isPageLoaded = !visiblePageElements.some(pageElement =>
+      !pageElement.isLoaded && !pageElement.hasFailed);
 
-      if (!canPageBeShown) {
-        canPageBeShown = pageElement.canBeShown;
-      }
-    });
+    const canPageBeShown = visiblePageElements.some(pageElement =>
+      pageElement.canBeShown);
```

Comment:
Minor: This is more declarative but requires iterating over `visiblePageElements` 3x instead of once. Recommend reverting.

Reply:
How about a compromise? :)